src/test/java/com/myroslav/cosmickitties/service/ProductServiceTest.java

```diff
@@ -0,0 +1,144 @@
+package com.myroslav.cosmickitties.service;
+
+import com.myroslav.cosmickitties.domain.Product;
+import com.myroslav.cosmickitties.dto.ProductDTO;
+import com.myroslav.cosmickitties.exception.ResourceNotFoundException;
+import com.myroslav.cosmickitties.mapper.ProductMapper;
+import com.myroslav.cosmickitties.repository.ProductRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Unit tests for ProductService.
+ * Use Mockito for repo + mapper, test happy and error flows.
+ */
+class ProductServiceTest {
+
+    private ProductRepository repo;
+    private ProductMapper mapper;
+    private ProductService service;
+
+    @BeforeEach
+    void setUp() {
+        repo = mock(ProductRepository.class);
```

Comment:
same please use mockito extension, or @SprinhBootTest. Because here you just keep re-creating the product service brand new instanse for each test